components/ai_chat/core/browser/ollama/ollama_model_fetcher_unittest.cc

```diff
@@ -0,0 +1,347 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/browser/ollama/ollama_model_fetcher.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/run_loop.h"
+#include "base/test/bind.h"
+#include "base/test/run_until.h"
+#include "base/test/task_environment.h"
+#include "brave/components/ai_chat/core/browser/model_service.h"
+#include "brave/components/ai_chat/core/browser/ollama/ollama_service.h"
+#include "brave/components/ai_chat/core/common/mojom/ollama.mojom.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "components/os_crypt/sync/os_crypt_mocker.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace ai_chat {
+
+namespace {
+
+using ::testing::_;
+
+// Mock OllamaService for testing OllamaModelFetcher
+class MockOllamaService : public OllamaService {
+ public:
+  explicit MockOllamaService(
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+      : OllamaService(url_loader_factory) {}
+
+  void FetchModels(ModelsCallback callback) override {
+    fetch_models_callback_ = std::move(callback);
+  }
+
+  void ShowModel(const std::string& model_name,
+                 ModelDetailsCallback callback) override {
+    show_model_callbacks_[model_name] = std::move(callback);
+  }
+
+  // Test helpers to trigger callbacks
+  void TriggerFetchModelsCallback(
+      std::optional<std::vector<ModelInfo>> models) {
+    if (fetch_models_callback_) {
+      std::move(fetch_models_callback_).Run(std::move(models));
+    }
+  }
+
+  void TriggerShowModelCallback(const std::string& model_name,
+                                std::optional<ModelDetails> details) {
+    auto it = show_model_callbacks_.find(model_name);
+    if (it != show_model_callbacks_.end()) {
+      std::move(it->second).Run(std::move(details));
+      show_model_callbacks_.erase(it);
+    }
+  }
+
+ private:
+  ModelsCallback fetch_models_callback_;
+  std::map<std::string, ModelDetailsCallback> show_model_callbacks_;
+};
```

Comment:
```suggestion
  class MockOllamaService : public OllamaService {
   public:
    explicit MockOllamaService(
        scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
        : OllamaService(url_loader_factory) {}

    MOCK_METHOD(void, FetchModels, (ModelsCallback), (override));
    MOCK_METHOD(void, ShowModel,
                (const std::string&, ModelDetailsCallback),
                (override));
  };
```
and use `EXPECT_CALL` and `WillOnce` to give the expected result.

Reply:
Fixed in 5db664f6e2640df6f1f59439dcfaa1d4b7ffd7a8